optuna/visualization/matplotlib/_slice.py

```diff
@@ -127,19 +127,29 @@ def _generate_slice_subplot(
     target_name: str,
 ) -> "PathCollection":
     ax.set(xlabel=subplot_info.param_name, ylabel=target_name)
-    x_values = subplot_info.x
     scale = None
     if subplot_info.is_log:
         ax.set_xscale("log")
         scale = "log"
-    elif not subplot_info.is_numerical:
-        x_values = [str(x) for x in subplot_info.x]
+    if subplot_info.is_numerical:
+        x_values = subplot_info.x
+        y_values = subplot_info.y
+        c_values = subplot_info.trial_numbers
+    else:
+        x_values = []
+        y_values = []
+        c_values = []
+        assert subplot_info.x_labels is not None
+        for x_label in subplot_info.x_labels:
+            for x, y, number in zip(subplot_info.x, subplot_info.y, subplot_info.trial_numbers):
+                if x == x_label:
+                    x_values.append(str(x))
+                    y_values.append(y)
+                    c_values.append(number)
```

Comment:
I think this double loop can be separated with `defaultdict(list)` or whatever. Are there any reason to keep this double loop?

Reply:
I couldn't understand your comment, could you please write me a suggested code?

Reply:
What I imegined was like this. 
```suggestion
        y_values_dict = defaultdict(list)
        c_values_dict = defaultdict(list)
        assert subplot_info.x_labels is not None
        for x, y, number in zip(subplot_info.x, subplot_info.y, subplot_info.trial_numbers):
            y_values_dict[x].append(y)
            c_values_dict[x].append(number)
        x_values_count = Counter(subplot_info.x)

        for x_label in subplot_info.x_labels:
            x_values.extend([str(x_label)] * x_values_count[x_label])
            y_values.extend(y_values_dict[x_label])
            c_values.extend(c_values_dict[x_label])
```

Reply:
Thank you. I have updated to eliminate unnecessary searching. How about it?

Reply:
Thank for the update. LGTM.